src/opt/ImproveMemoryOps.cpp

```diff
@@ -653,226 +653,155 @@ static bool lOffsets32BitSafe(llvm::Value **offsetPtr, llvm::Instruction *insert
-                                       : __pseudo_scatter_factored_base_offsets64_double,
-               g->target->useScatter() ? __pseudo_scatter_base_offsets32_double
-                                       : __pseudo_scatter_factored_base_offsets32_double,
-               false, false),
-        GSInfo(__pseudo_prefetch_read_varying_1,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_1_native : __prefetch_read_varying_1,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_1_native : __prefetch_read_varying_1, false,
-               true),
-
-        GSInfo(__pseudo_prefetch_read_varying_2,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_2_native : __prefetch_read_varying_2,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_2_native : __prefetch_read_varying_2, false,
-               true),
-
-        GSInfo(__pseudo_prefetch_read_varying_3,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_3_native : __prefetch_read_varying_3,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_3_native : __prefetch_read_varying_3, false,
-               true),
-
-        GSInfo(__pseudo_prefetch_read_varying_nt,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_nt_native : __prefetch_read_varying_nt,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_read_varying_nt_native : __prefetch_read_varying_nt,
-               false, true),
-
-        GSInfo(__pseudo_prefetch_write_varying_1,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_1_native : __prefetch_write_varying_1,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_1_native : __prefetch_write_varying_1,
-               false, true),
-        GSInfo(__pseudo_prefetch_write_varying_2,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_2_native : __prefetch_write_varying_2,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_2_native : __prefetch_write_varying_2,
-               false, true),
-
-        GSInfo(__pseudo_prefetch_write_varying_3,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_3_native : __prefetch_write_varying_3,
-               g->target->hasVecPrefetch() ? __pseudo_prefetch_write_varying_3_native : __prefetch_write_varying_3,
-               false, true),
+// Contains a function call substitution info for gather/scatter/prefetch calls.
+// We construct this class oblects in program constructors phase, i.e., before initialization of m->module, g->target
+// and any of LLVMTypes values. Moreover, g->target values are not same during multi-target compilation. Because of
+// that, we postpone evaluation of types, choosing the proper function before actual call replacement here and in
+// similar types across this file.
+struct GSInfo {
```

Comment:
Any specific reason why it's a struct, but not a class? Some fields and functions are private, so probably it makes sense to make them private.

Reply:
No reason, it just was so. Fields are private now.